### PR TITLE
fix: allow default IoT Hub SKUs (F1)

### LINKED ISSUE #534: Running `iotedgedev iothub setup` with all default values results in error
- iotedgedev Version: 3.3.4
- Python Version: 3.9.12
- Pip Version: 22.0.4 
- Development machine OS Version: Windows WSL, iotedgedev devcontainer
- IoT Edge device OS Version: -

### Steps to Reproduce:

1. Not logged in into azure cli
3. run `iotedgedev iothub setup` 
4. select subscription (ms internal subscription) // not sure if this made a difference

### Expected

Create iot hub with the selected default values

### Actual

```sh
=================================
======== SETTING IOT HUB ========
=================================

Setting IoT Hub to 'iotedgedev-iothub-d278b7'...


Checking if 'iotedgedev-iothub-d278b7' IoT Hub exists...

Could not locate the iotedgedev-iothub-d278b7 in iotedgedev-rg.
Creating 'iotedgedev-iothub-d278b7' in 'iotedgedev-rg' with 'F1' sku...

Creating IoT Hub. Please wait as this could take a few minutes to complete...
ERROR: Could not create the IoT Hub iotedgedev-iothub-d278b7 in iotedgedev-rg with sku F1.

ERROR: ERROR: (400016) Invalid PartitionCount 4 - value must be between 2 and 2. If you contact a support representative please include this correlation identifier: cf2e50d2-0ba3-4e2f-b10d-12e38678e6de, timestamp: 2022-04-14 09:43:45Z, errorcode: IH400016.
Code: 400016
Message: Invalid PartitionCount 4 - value must be between 2 and 2. If you contact a support representative please include this correlation identifier: cf2e50d2-0ba3-4e2f-b10d-12e38678e6de, timestamp: 2022-04-14 09:43:45Z, errorcode: IH400016.

Error: Could not create IoT Hub iotedgedev-iothub-d278b7 in iotedgedev-rg
Enter the IoT Hub Name (Creates a new IoT Hub if not found): [iotedgedev-iothub-d278b7]: ^[[A^[[B^[[B^[[B^[[B^[[B^[[B
```

### PR DESCRIPTION
**Pull request by roel4ez**
## Checklist

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] Versions update reflected in all places (both __init__.py files, CHANGELOG, setup.py, setup.cfg)
- [x] Unit tests pass locally
- [x] Quickstart steps locally validated
- [ ] Passes unit tests in CICD pipeline (green on Github pipeline)
- [ ] Pypi RC version passes Edge CICD pipeline validation for cross-tool validation
- [ ] Pull request includes test coverage for the included changes
- [x] Documentation is updated for the included changes

### General guidelines

- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.

## Description

Fixes #534

Creating a F1 IoT Hu requires setting the partition count to 2. The default from the CLI is 4, so in case F1 is requested the partition count is added to the command.

### CODE REVIEW

File: iotedgedev/azurecli.py [hunk 1]
```diff
@@ -465,9 +465,11 @@ def create_iothub(self, value, resource_group, sku):
                 self.output.prompt(
                     "Creating IoT Hub. Please wait as this could take a few minutes to complete...")
 
-                result = self.invoke_az_cli_outproc(["iot", "hub", "create", "--name", value, "--resource-group",
-                                                     resource_group, "--sku", sku, "--query", "[].{\"IoT Hub\":name}", "--out", "table"],
-                                                    f("Could not create the IoT Hub {value} in {resource_group} with sku {sku}."), stdout_io=io, stderr_io=error_io)
+                cmd = ["iot", "hub", "create", "--name", value, "--resource-group", resource_group, "--sku", sku, "--query", "[].{\"IoT Hub\":name}", "--out", "table"]
+                if sku == "F1":
+                    cmd = cmd + ["--partition-count", "2"]
```
Comment (marianan) [inline iotedgedev/azurecli.py:470]: >"2" [](http://example.com/codeflow?start=54&length=3)

could you please add a comment why was the default value set to 2? Documenting some context here will help with maintaining the code int he future.

Comment (roel4ez) [reply iotedgedev/azurecli.py:470]: I can't locate the documentation where it is specified unfortunately. However, to verify this you can check the Azure Portal: when trying to create an F1 IoT Hub, it does not allow you to change the partition count on the Management Tab
image

Comment (marianan) [reply iotedgedev/azurecli.py:470]: I see, thanks Roel for clarifying, Could you please add a comment for context here that says that the value is chosen for parity with Portal experience.

**Review by marianan — APPROVED** [review]
:shipit:

**Pull request merged**